### PR TITLE
Merge Project

### PR DESCRIPTION
**Pull request by DevaWinner**
Hello Reviewer,

Please kindly review my project

## Design

Implemented following the design guidelines, including colors, typography, and layout as specified in the provided design. 

## Interactions

- **Splash Screen:**
  - Displays the app name with links to sign up and log in pages.
- **Sign Up and Log In Pages:**
  - User registration with full name, email, and password.
  - User login using email and password.
- **Home Page (Categories Page):**
  - Displays categories with name, icon, and total transaction amount.
  - Clicking on a category navigates to the transactions page for that category.
  - "Add a new category" button at the bottom for category creation.
- **Transactions Page:**
  - Lists transactions for a category, ordered by most recent.
  - Displays total category amount at the top.
  - "Add a new transaction" button for transaction creation.
  - "Back" button (<) navigates back to the home page.
- **"Add a New Category" Page:**
  - Form for creating a new category with name and icon.
  - "Save" button creates the new category and navigates to the home page on success.
  - "Back" button (<) navigates back to the home page.
- **"Add a New Transaction" Page:**
  - Form for creating a new transaction with name, amount, and category.
  - "Save" button creates the new transaction and navigates to the transactions page for that category.
  - "Back" button (<) navigates back to the transactions page for that category.

## Testing

Comprehensive unit and integration tests cover all vital components of the Ruby on Rails application.

## Technical Implementation

- Database: Postgres
- Authentication: Devise
- Input Validation: Implemented to ensure security.
- Deployment: Online and accessible.
- Database Schema: Reflects the specified structure.

## Documentation and Code Quality

- Linter Errors: Checked and resolved.
- Gitflow: Followed correct gitflow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,147 @@
...
+### Run Test
+To run the testing, execute the following command:
+```
+rspec spec --format documentation
+```
+
+- [Live Demo](https://expenditure-2i8a.onrender.com/)
+
+- [Project Presentation](https://www.loom.com/share/68f477522af04d1c9a51f321866b5fa6?sid=ac1d8821-7d79-4c57-bc47-f72894e8d4e7)
+
```
Comment (Gambit142) [inline README.md]: - [ ] I tried several times and on different browsers to play your video recording but it wouldn't play. Kindly record a video that is playable and meets all the requirements 👍 😎

File: db/migrate/20231016095850_change_column_name_in_expenses.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class ChangeColumnNameInExpenses < ActiveRecord::Migration[7.1]
+  def change
+    rename_column :expenses, :author_id_id, :author_id
+  end
+end
```
Comment (Gambit142) [inline db/migrate/20231016095850_change_column_name_in_expenses.rb]: - [ ] When I try to run your migration files, I get these errors:

image

In the `Expenses` table, you don't have any column named `author_id_id`. The references to the `Users` table will create a `user_id` column in the `Expenses` table. Kindly delete this migration for a seamless running of all migration files. 👍 😎

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      </div>
+    <% end %>
+
+    <div class="form-group">
+      <%= f.submit "Log in", class: "btn btn-primary" %>
+    </div>
+  <% end %>
+  
+  <%= render "devise/shared/links" %>
+</div>
```
Comment (Gambit142) [inline app/views/devise/sessions/new.html.erb]: - [ ] When I start your application, I am directed to this page, which shouldn't be the case. The requirements state that you create a **_Splash screen_** that would be the first thing a user sees when they start your application. Kindly implement this requirement. 👍 🖱️ 

---

#### The image below shows the design of the Splash Screen. 

image

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          </div>
+        </div>
+      <% end %>
+    <% end %>
+  </div>
+  <br/>
+  <div class="p-2 d-flex justify-content-center" style="background-color: #5fb523">
+    <% if user_signed_in? %>
+      <%= link_to "Add new group", new_group_path, class: "btn btn-outline-light" %>
+    <% else %>
```
Comment (Gambit142) [inline app/views/groups/index.html.erb]: - [ ] The **_Add new group button_** should be positioned at the bottom of the page based on the design. It is essential to follow design guidelines when creating applications. Kindly implement this change.


| Your Design  | Behance Design |
| ------------- | ------------- |
| image  | image  |

#### N.B: Please adjust the spacing between the **_white border line_** and the **_Add new group text_**. You could use CSS padding to achieve this. Kindly refer to the Behance design layout attached to this comment to understand how the button should look. 👍 😎

File: app/views/groups/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  <br />
+  <div class="p-2 d-flex justify-content-between" style="background-color: #5fb523; position: fixed; bottom: 0; width: 100vw;">
+    <div>
+      <%= form.submit "Create Group", class: "btn btn-outline-light" %>
+    </div>
+    <div>
+      <%= link_to "Back to Groups", groups_path, class: "btn btn-outline-light" %>
+    </div>
+  </div>
+<% end %>
```
Comment (Gambit142) [inline app/views/groups/new.html.erb]: - [ ] I understand that programmers love being creative with designs. However, it is paramount that designers stick to design guidelines in projects. This shows utmost professionalism and attention to detail. On this page, I would recommend you do the following ✍️,

- Move the **_Back to groups button_** to the top of the page in the navigation section (You don't need to change it to an icon, just position it at the right corner of the navigation section. 😃 
- Make the _**Create Group button**_ look identical to that of the Behance design layout for buttons on their templates. 🖥️  

--- 

| Your Design  | Behance Design |
| ------------- | ------------- |
| image  | image  |

File: app/views/expenses/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    <div class="p-2 d-flex justify-content-between" style="background-color: #5fb523; position: fixed; bottom: 0; width: 100vw;">
+      <div style="margin-right: 2rem">
+        <%= link_to "Add expense", new_group_expense_path(@group), class: "btn btn-outline-light" %>
+      </div>
+      <div>
+        <%= link_to "Back to groups", groups_path, class: "btn btn-outline-light" %>
+      </div>
+    </div>
+  <% end %>
+</div>
```
Comment (Gambit142) [inline app/views/expenses/index.html.erb]: - [ ] I understand that programmers love being creative with designs. However, it is paramount that designers stick to design guidelines in projects. This shows utmost professionalism and attention to detail. On this page, I would recommend you do the following ✍️,

- Move the **_Back to groups button_** to the top of the page in the navigation section (You don't need to change it to an icon, just position it at the right corner of the navigation section. 😃 
- Make the _**Add expense**_ look identical to that of the Behance design layout for buttons on their templates. 🖥️  

--- 

| Your Design  | Behance Design |
| ------------- | ------------- |
| image  | image  |

File: app/views/expenses/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  <div class="p-2 d-flex justify-content-between" style="background-color: #5fb523; position: fixed; bottom: 0; width: 100vw;">
+    <div style="margin-right: 2rem">
+      <%= f.submit "Create expense", class: "btn btn-outline-light" %>
+    </div>
+    <div>
+      <%= link_to "Back to Expenses", group_expenses_path(group_id: params[:group_id]), class: "btn btn-outline-light" %>
+    </div>
+  </div>
+<% end %>
```
Comment (Gambit142) [inline app/views/expenses/new.html.erb]: - [ ] I understand that programmers love being creative with designs. However, it is paramount that designers stick to design guidelines in projects. This shows utmost professionalism and attention to detail. On this page, I would recommend you do the following ✍️,

- Move the **_Back to groups button_** to the top of the page in the navigation section (You don't need to change it to an icon, just position it at the right corner of the navigation section. 😃 
- Make the _**Create expense**_ look identical to that of the Behance design layout for buttons on their templates. 🖥️  

--- 

| Your Design  | Behance Design |
| ------------- | ------------- |
| image  | image  |

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @DevaWinner 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Created controllers ✔️ 
- Created models ✔️ 
- Created routes ✔️ 
- Wrote unit and integration tests ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: db/migrate/20231016095850_change_column_name_in_expenses.rb (hunk 2 above)
Comment (DevaWinner) [reply db/migrate/20231016095850_change_column_name_in_expenses.rb]: Thank you. I have resolved this

File: app/views/devise/sessions/new.html.erb (hunk 3 above)
Comment (DevaWinner) [reply app/views/devise/sessions/new.html.erb]: I have restructured app interface and including splash screen.

File: app/views/groups/new.html.erb (hunk 5 above)
Comment (DevaWinner) [reply app/views/groups/new.html.erb]: I have restructured app interface

File: app/views/expenses/new.html.erb (hunk 7 above)
Comment (DevaWinner) [reply app/views/expenses/new.html.erb]: I have restructured app interface

File: README.md (hunk 1 above)
Comment (DevaWinner) [reply README.md]: I have done another recording

Comment (Abenezer-Tilahun) [reply README.md]: Fixed.

File: db/migrate/20231016095850_change_column_name_in_expenses.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply db/migrate/20231016095850_change_column_name_in_expenses.rb]: Fixed.

File: app/views/expenses/new.html.erb (hunk 7 above)
Comment (Abenezer-Tilahun) [reply app/views/expenses/new.html.erb]: Fixed.

File: app/views/expenses/index.html.erb (hunk 6 above)
Comment (Abenezer-Tilahun) [reply app/views/expenses/index.html.erb]: Fixed.

File: app/views/groups/new.html.erb (hunk 5 above)
Comment (Abenezer-Tilahun) [reply app/views/groups/new.html.erb]: Fixed.

File: app/views/groups/index.html.erb (hunk 4 above)
Comment (Abenezer-Tilahun) [reply app/views/groups/index.html.erb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @DevaWinner 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

🎲 Nice job acknowledging Gregoire Vella, creator of the original design
🎲 Properly added required models, controllers, and corresponding views
🎲 Devise correctly implemented for authentication!
🎲 Schema follow given ERD!
🎲 Correctly deployed app!
🎲 Awesome UI, closely matches given template design!
🎲 App is running as intended!
🎲 Added required spec test cases and are passing!
🎲 ReadMe looks great!
🎲 Good job for Implemented Suggested changes
 

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

**Pull request merged**